src/app/icd/client/ICDClientStorage.h

```diff
@@ -37,22 +36,13 @@ namespace app {
 class ICDClientStorage
 {
 public:
-    using ICDClientInfoIterator = CommonIterator<ICDClientInfo>;
-
     virtual ~ICDClientStorage() = default;
 
-    /**
-     * Iterate through persisted ICD Client Info
-     *
-     * @return A valid iterator on success. Use CommonIterator accessor to retrieve ICDClientInfo
-     */
-    virtual ICDClientInfoIterator * IterateICDClientInfo() = 0;
```

Comment:
So how will a user know after reboot which entries are available if this API is removed?

Reply:
I may not realize the user case, why application need to iterate entries? in which scenario?

Reply:
I may not understand how the API is supposed to be used, but I imagined that after a reboot an ICD would need to recreate ICD sender structures to know where to send the check-in messages. If the `GetEntry` takes a peer node ID, how do I know all the peer node IDs that I should query?

Reply:
@Damian-Nordic This is client-side API, not server-side.  The recipient of check-in messages, not the sender.

Reply:
Ah, sorry for the confusion!